python_scripts/metrics_regression.py

```diff
@@ -185,21 +202,33 @@
 model_transformed_target.fit(data_train, target_train)
 target_predicted = model_transformed_target.predict(data_test)
 
-# %%
-predicted_actual = {
-    "True values (k$)": target_test, "Predicted values (k$)": target_predicted}
-predicted_actual = pd.DataFrame(predicted_actual)
+fig, axs = plt.subplots(ncols=2, figsize=(13, 5))
 
-# %%
-sns.scatterplot(data=predicted_actual,
-                x="True values (k$)", y="Predicted values (k$)",
-                color="black", alpha=0.5)
-plt.axline((0, 0), slope=1, label="Perfect fit")
-plt.axis('square')
-plt.legend()
-_ = plt.title("Regression using a model that\n transform the target before "
-              "fitting")
+PredictionErrorDisplay.from_predictions(
+    y_true=target_test,
+    y_pred=target_predicted,
+    kind="actual_vs_predicted",
+    scatter_kwargs={"alpha": 0.5},
+    ax=axs[0]
+)
+axs[0].axis("square")
+axs[0].set_xlabel("Predicted values (k$)")
+axs[0].set_ylabel("True values (k$)")
+
+PredictionErrorDisplay.from_predictions(
+    y_true=target_test,
+    y_pred=target_predicted,
+    kind="residual_vs_predicted",
+    scatter_kwargs={"alpha": 0.5},
+    ax=axs[1]
+)
+axs[1].axis("square")
+axs[1].set_xlabel("Predicted values (k$)")
+axs[1].set_ylabel("Residual values (k$)")
+
+_ = fig.suptitle("Regression using a model that\ntransforms the target "
+                 "before fitting", y=1.1)
 
 # %% [markdown]
 # Thus, once we transformed the target, we see that we corrected some of the
-# high values.
+# predictions of the high values.
```

Comment:
The model with the transformed target seems to exhibit fewer structure in its residuals: over-estimation and under-estimation errors seems to be more balanced.

We can confirm this by computing the previous metrics for that last model and observe that they all improved w.r.t. the linear regression model without the target transformation.

TODO: add new cells to compute all the metrics for the last model.

While a common practice, performing such a target transformation for linear regression is often frowned upon by statisticians: it is often mathematically more
justified to instead adapt the loss function of the regression model itself, for
instance by fitting a [PoissonRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.PoissonRegressor.html) or a [TweedieRegressor](
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.TweedieRegressor.html) model instead of `LinearRegression`. In particular those
models indeed use an internal "log link" function that makes them more suited for
this kind of positive-only target data distributions but this analysis is beyond the scope of this MOOC.

The interested readers are encouraged to learn more about those models, in particular by reading the user guide and the examples linked for their API documentation pages above.